### PR TITLE
Various deprecation fixes and updates for 2018.3

### PR DESCRIPTION
**Pull request by Kerfuffles**
## Changes
* `PrefabUtility.GetPrefabObject(Object prefab) `-> `PrefabUtility.GetPrefabInstanceHandle(Object prefab)`
* Updated #if directives to account for unity 2018.3 changes

## Notes
I'm waiting on unity tech to drop the official release before saying this is ready to merge, just in case there are more deprecations or changes that may break things in the future.

### CODE REVIEW

**Comment by Kerfuffles** [comment]
image
Discovered that putting the 2D shape editor in full screen in 2018.3 causes vertices and line segments to be offset. It doesn't seem to be an issue when its in a normal window - only when docked, and shift + space is pressed. @Henry00IS

**Comment by Kerfuffles** [comment]
[PreferenceItem(string name)] -> [SettingsProvider] which the implementation of can be found here: https://github.com/Unity-Technologies/UnityCsReference/blob/master/Editor/Mono/Inspector/GraphicsSettingsInspector.cs

The attribute and the implementation of it is currently undocumented. Will update with more information when it's available.

File: Scripts/Tools/Tool.cs [hunk 1]
```diff
@@ -62,12 +65,16 @@ public void SetSelection(GameObject selectedGameObject, GameObject[] selectedGam
 
 			// Make sure it's not null and that it isn't a prefab in the project
 			if(selectedGameObject != null
-#if UNITY_2018_2
+#if UNITY_2018_2_OR_NEWER || UNITY_2018_3
```
Comment (Henry00IS) [inline Scripts/Tools/Tool.cs]:
```suggestion
#if UNITY_2018_2_OR_NEWER
```

I think you missed removing a " || UNITY_2018_3" statement. :)

**Comment by Henry00IS** [comment]
Brilliant work! Appears to have fixed all of the warnings in 2018! 😁

**Pull request merged**